lib/ansible/modules/cloud/vmware/vmware_guest_disk.py

```diff
@@ -0,0 +1,631 @@
+                    current_disk['datastore'] = datastore
+                    current_disk['autoselect_datastore'] = False
+                elif 'autoselect_datastore' in disk:
+                    # Find datastore which fits requirement
+                    datastores = get_all_objs(self.content, [vim.Datastore])
+                    if not datastores:
+                        self.module.fail_json(msg="Failed to gather information about"
+                                                  " available datastores in given datacenter.")
+                    datastore = None
+                    datastore_freespace = 0
+                    for ds in datastores:
+                        if ds.summary.freeSpace > datastore_freespace:
+                            # If datastore field is provided, filter destination datastores
+                            datastore = ds
+                            datastore_freespace = ds.summary.freeSpace
+                    current_disk['datastore'] = datastore
+
+                if 'datastore' not in disk and 'autoselect_datastore' not in disk:
+                    self.module.fail_json(msg="Either 'datastore' or 'autoselect_datastore' is"
+                                              " required parameter while creating disk for "
+                                              "disk index [%s]." % disk_index)
+
+                if [x for x in disk.keys() if x.startswith('size_') or x == 'size']:
+                    # size, size_tb, size_gb, size_mb, size_kb
+                    disk_size_parse_failed = False
+                    if 'size' in disk:
+                        size_regex = re.compile(r'(\d+(?:\.\d+)?)([tgmkTGMK][bB])')
+                        disk_size_m = size_regex.match(disk['size'])
+                        if disk_size_m:
+                            expected = disk_size_m.group(1)
+                            unit = disk_size_m.group(2)
+                        else:
+                            disk_size_parse_failed = True
+                        try:
+                            if re.match(r'\d+\.\d+', expected):
+                                # We found float value in string, let's typecast it
+                                expected = float(expected)
+                            else:
+                                # We found int value in string, let's typecast it
+                                expected = int(expected)
+                        except (TypeError, ValueError, NameError):
+                            disk_size_parse_failed = True
+
+                        if disk_size_parse_failed:
+                            # Common failure
+                            self.module.fail_json(msg="Failed to parse disk size for disk index [%s],"
+                                                      " please review value provided"
+                                                      " using documentation." % disk_index)
```

Comment:
You added NameError at the except list, so there is no need for disk_size_parse_failed.

```
+                        if disk_size_m:
+                            expected = disk_size_m.group(1)
+                            unit = disk_size_m.group(2)
+                        try:
+                            if re.match(r'\d+\.\d+', expected):
+                                # We found float value in string, let's typecast it
+                                expected = float(expected)
+                            else:
+                                # We found int value in string, let's typecast it
+                                expected = int(expected)
+                        except (TypeError, ValueError, NameError):
+                            # Common failure
+                            self.module.fail_json(msg="Failed to parse disk size for disk index [%s],"
+                                                      " please review value provided"
+                                                      " using documentation." % disk_index)
```

Reply:
also, take a look at https://github.com/ansible/ansible/pull/36617#discussion_r170312630

Reply:
also, have a look at https://github.com/ansible/ansible/pull/36617#discussion_r170312630